gpg-expiration.py

```diff
@@ -49,8 +51,21 @@ def load_key(key_id):
             raise ValueError('Unable to load key', key_id) from ex
 
 
+def check_all(not_before):
```

Comment:
> Similarly, I wanted to make sure the `check_expiration` function works identically as before, rather than making separate `check_single` and `check_all` functions.

I agree with not wanting separate `check_single` and `check_all` functions, but I think the best way to achieve that goal is to merge them into a single `check_multiple(patterns, not_before)`. That might also more directly match your use case, although I think it is equally useful in both the `--all` and the multiple `--key-id` approaches - and should be compatible with both.

I imagine this new function would probably accept a list of patterns to pass to `keylist`, with a single-element argument containing `None` in the case of `--all`, like `check_multiple([None], not_before)`.

Does that make sense? Would you be interested in implementing that in this PR?

Reply:
Makes sense in theory. I'll need to start making changes to think about what it ends up looking like. I'm fine updating this PR to make those changes.